app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt

```diff
@@ -0,0 +1,122 @@
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import androidx.recyclerview.widget.RecyclerView
+import dalalstreet.api.models.DailyChallengeOuterClass
+import dalalstreet.api.models.UserStateOuterClass
+import org.jetbrains.anko.textColor
+import org.pragyan.dalal18.R
+import org.pragyan.dalal18.databinding.ChallengeListItemBinding
+
+class DailyChallengesRecyclerAdapter(
+        private val dailyChallenges: MutableList<DailyChallengeOuterClass.DailyChallenge>,
+        private val userStates: MutableList<UserStateOuterClass.UserState>,
+        private val checkUserStateListener: CheckUserStateListener
+) : RecyclerView.Adapter<DailyChallengesRecyclerAdapter.DailyChallengeViewHolder>() {
+    inner class DailyChallengeViewHolder(val binding: ChallengeListItemBinding): RecyclerView.ViewHolder(binding.root)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DailyChallengeViewHolder {
+        return DailyChallengeViewHolder(ChallengeListItemBinding.inflate(LayoutInflater.from(parent.context),parent,false))
+    }
+
+    override fun getItemCount(): Int = dailyChallenges.size
+
+    override fun onBindViewHolder(holder: DailyChallengeViewHolder, position: Int) {
+//
+        Log.i("daily",dailyChallenges[position].toString())
+        val challengeId = dailyChallenges[position].challengeId
+        val challengeType = dailyChallenges[position].challengeType
+        val value = dailyChallenges[position].value
+        val rewardAmount = dailyChallenges[position].reward
+
+
+        if(challengeType.equals("Cash")||challengeType.equals("NetWorth")||challengeType.equals("StockWorth"))
+            holder.binding.challengeText.text = "Increase your ${challengeType} by ${value}"
+        else{
+            val stockId = dailyChallenges[position].stockId
+            val companyName = checkUserStateListener.getCompanyNameFromStockId(stockId)
+            holder.binding.challengeText.text = "Increase the number of stocks of ${companyName} by ${value}"
+        }
+        holder.binding.rewardText.text = "\uD83D\uDCB5 $rewardAmount"
+        if(checkUserStateListener.isDailyChallengeOpen()){
+            val initialValue = userStates[position].initialValue
+            holder.binding.progressTextView.visibility = View.VISIBLE
+
+            holder.binding.progressImage.visibility=View.GONE
+            holder.binding.claimRewardButton.visibility=View.GONE
+            var progress:Long=0
+            when(challengeType){
+                "Cash"->{
+                    val currentCash = checkUserStateListener.getCashWorth()
+                    progress = currentCash - initialValue
+
+                }
+                "NetWorth"->{
+                    val netWorth = checkUserStateListener.getNetWorth()
+                     progress = netWorth - initialValue
+                }
+                "StockWorth"->{
```

Comment:
Make "Cash" "NetWorth" as companion objects and then use it everywhere

Reply:
companion objects of dalalViewModel??

Reply:
No, of this class itself